### PR TITLE
Fix #157, replace deprecated SqlContext with SparkSession

### LINKED ISSUE #157: Remove deprecated SqlContext, replace it with SparkSession
class SqlContext is deprecated in Spark 2.0. We should use class SparkSession instead.

### PR DESCRIPTION
**Pull request by clockfly**
This PR replaces deprecated SqlContext with class SparkSession.

Fix #157

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -209,29 +212,29 @@ df.write.options(parameters).avro("/tmp/output")
...
-DataFrame df = sqlContext.read().format("com.databricks.spark.avro")
-    .load("src/test/resources/episodes.avro");
+Dataset<Row> df = spark.read().format("com.databricks.spark.avro")
+  .load("src/test/resources/episodes.avro");
 
 // Saves the subset of the Avro records read in
-df.filter($"age > 5").write()
-    .format("com.databricks.spark.avro")
-    .save("/tmp/output");
+df.filter(functions.expr("doctor > 5")).write()
```
Comment (cloud-fan) [inline README.md]: `functions` is already imported.

Comment (cloud-fan) [inline README.md]: why change `age` to `doctor`?

File: src/test/scala/com/databricks/spark/avro/AvroSuite.scala [hunk 2]
```diff
@@ -30,38 +30,37 @@ import org.apache.avro.file.DataFileWriter
...
   val episodesFile = "src/test/resources/episodes.avro"
   val testFile = "src/test/resources/test.avro"
 
-  private var sqlContext: SQLContext = _
+  private var spark: SparkSession = _
 
   override protected def beforeAll(): Unit = {
     super.beforeAll()
-    sqlContext = new SQLContext(new SparkContext("local[2]", "AvroSuite"))
+    spark = SparkSession.builder().master("local").appName("AvroSuite").getOrCreate()
```
Comment (cloud-fan) [inline src/test/scala/com/databricks/spark/avro/AvroSuite.scala]: why change `local[2]` to `local`?

Comment (clockfly) [inline src/test/scala/com/databricks/spark/avro/AvroSuite.scala]: I think there is not much difference. but I can change it back.

File: README.md (hunk 1 above)
Comment (clockfly) [inline README.md]: Because the age field doesn't exist in test file. The old code example is wrong.

Comment (clockfly) [inline README.md]: @cloud-fan 

This is a java code example. functions need to be imported explicitly.

**Comment by cloud-fan** [comment]
LGTM

**Comment by clockfly** [comment]
I just did a squash.

**Pull request closed without merging**